### ISSUE TITLE
Enable commonLabels for all pushProxy template to allow more Prometheus scraping

### ISSUE THREAD

**Issue by atsai1220** [rancher/rancher#39427]
We'd like to use `prometheus.prometheusSpec.serviceMonitorSelector` to select a subset of `ServiceMonitors` within the cluster (including from other applications). 

In the `rancher-monitoring` chart, we can specify `commonLabels` across all resources.
https://github.com/rancher/charts/blob/c5260036905fc3d393655fb827e1becd88b760d3/charts/rancher-monitoring/16.6.1%2Bup16.6.0/values.yaml#L441-L443

However, the subchart for pushproxy does not honor these labels
https://github.com/rancher/charts/blob/c5260036905fc3d393655fb827e1becd88b760d3/charts/rancher-monitoring/16.6.1%2Bup16.6.0/charts/rkeControllerManager/templates/_helpers.tpl#L41-L45

Enabling `podMonitorSelectorNilUsesHelmValues: true` allows Prometheus to select all expected ServiceMonitors with the key `release`. However that means other applications in the cluster must use the key `release` but that may conflict with Helm installations.

It seems we would only need to update each subchart to have something like this in the their `_helpers.tpl`
``` yaml
{{- define "pushProxy.commonLabels" -}}
release: {{ .Release.Name }}
component: {{ .Values.component | quote }}
provider: kubernetes
{{- if .Values.commonLabels}}
{{ toYaml .Values.commonLabels }}
{{- end }}
{{- end -}}
```

Then the entire `rancher-monitoring` chart's values can be:
```
rkeControllerManager:
  commonLabels:
    owner: team-a
```

In summary, the labels for `kube-prometheus-stack.labels` does not propagate down to the charts used by Rancher for push proxy.
https://github.com/rancher/charts/blob/c5260036905fc3d393655fb827e1becd88b760d3/charts/rancher-monitoring/16.6.1%2Bup16.6.0/templates/_helpers.tpl#L174-L186
